### PR TITLE
Display replicas, cores and nodes in sizing tool.

### PR DESCRIPTION
**Pull request by jeschkies**
**What this PR does / why we need it**:
No every users wants to deploy Loki with Helm. That's why we are going to display the required replicas as well.

image

This change also fixes two bugs in the algorithm and introduces some tests for its invariant.

**Checklist**
- [ ] Reviewed the `CONTRIBUTING.md` guide
- [x] Documentation added
- [x] Tests updated
- [ ] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: pkg/sizing/algorithm_test.go [hunk 1]
```diff
@@ -33,3 +33,14 @@ func Test_AlgorithTest_Algorith(t *testing.T) {
 		t.Error(err)
 	}
 }
+
+func Test_CoresNodeInvariant(t *testing.T) {
```
Comment (jeschkies) [inline pkg/sizing/algorithm_test.go:37]: @trevorwhitney I may misunderstand the logic of the algorithm but it fails here in that it demands more cores than is available on the nodes.

Comment (trevorwhitney) [reply pkg/sizing/algorithm_test.go:37]: That sounds like a bug. It should add additional nodes if there aren't enough cores. After a brief look at the code nothing jumps out, would have to dig in more.

File: pkg/sizing/algorithm.go [hunk 2]
```diff
@@ -33,7 +34,7 @@ func calculateClusterSize(nt NodeType, bytesDayIngest float64, qperf QueryPerf)
 	replicasOnLastNode := math.Mod(numWriteReplicasNeeded, writeReplicasPerNode)
 
 	coresOnLastNode := 0.0
-	if replicasOnLastNode >= 0.0 {
+	if replicasOnLastNode > 0.0 {
```
Comment (jeschkies) [inline pkg/sizing/algorithm.go:37]: This was the bug in the algorithm. Glad the test caught it.

File: pkg/sizing/algorithm_test.go [hunk 3]
```diff
@@ -32,4 +32,21 @@ func Test_AlgorithmTest_Algorithm(t *testing.T) {
 	if err := quick.Check(f, nil); err != nil {
 		t.Error(err)
 	}
+
+	// Sanity check for 1TB/Day
+	size := calculateClusterSize(NodeTypesByProvider["AWS"]["t2.xlarge"], 1e12, Basic)
+	require.Equalf(t, 4, size.TotalNodes, "given ingest=1PB/Day totla nodes must be big")
```
Comment (jeschkies) [inline pkg/sizing/algorithm_test.go:38]: This catches the other bug.

File: pkg/sizing/algorithm.go [hunk 4]
```diff
@@ -23,7 +24,7 @@ const (
 func calculateClusterSize(nt NodeType, bytesDayIngest float64, qperf QueryPerf) ClusterSize {
 
 	// 1 Petabyte per day is maximum. We use decimal prefix https://en.wikipedia.org/wiki/Binary_prefix
-	bytesDayIngest = math.Min(bytesDayIngest, 1e12)
+	bytesDayIngest = math.Min(bytesDayIngest, 1e15)
```
Comment (jeschkies) [inline pkg/sizing/algorithm.go:27]: This was also an issue.

File: pkg/sizing/algorithm.go (hunk 2 above)
Comment (trevorwhitney) [reply pkg/sizing/algorithm.go:37]: nice!

File: docs/sources/installation/sizing/index.md [hunk 5]
```diff
@@ -86,7 +103,8 @@ This tool helps to generate a Helm Charts `values.yaml` file based on specified
 </style>
 
 <script>
-const API_URL = `https://logql-analyzer.grafana.net/next/api/sizing`
+//const API_URL = `https://logql-analyzer.grafana.net/next/api/sizing`
```
Comment (trevorwhitney) [inline docs/sources/installation/sizing/index.md:106]: don't think you want to commit the localhost url

**Review by trevorwhitney — APPROVED** [review]
Looks great. I think there was an accidental localhost url committed, otherwise LGTM!

File: docs/sources/installation/sizing/index.md (hunk 5 above)
Comment (jeschkies) [reply docs/sources/installation/sizing/index.md:106]: Yes. Thanks for reminding me. This is for testing.

**Pull request merged**